File: day 15/linked_list.cs
using System;
class Node
{
	public int data;
	public Node next;
    public Node(int d){
        data=d;
        next=null;
    }
		
}
class Solution {

	public static Node insert(Node head,int data)
	{
        Node temp = new Node(data);
        if (head == null) { // empty list becomes the new node
            head = temp;
        } else { // find last and link the new node
            Node last = head;
            while(last.next != null) 
                last=last.next;
            last.next = temp;
        }
        return head;
    }

	public static void display(Node head)
	{
		Node start=head;
		while(start!=null)
		{
			Console.Write(start.data+" ");
			start=start.next;
		}
	}
    static void Main(String[] args) {
	
		Node head=null;
        int T=Int32.Parse(Console.ReadLine());
        while(T-->0){
            int data=Int32.Parse(Console.ReadLine());
            head=insert(head,data);
        }
		display(head);
	}
}

File: day 15/linked_list.py
"""Exploring linked lists in Python."""

class Node: # pylint: disable=too-few-public-methods
    """Node class to create the linked list."""

    def __init__(self, data):
        """Initialise the data and the next node."""
        self.data = data
        self.next = None

class Solution:
    """Display and Insert funcitons."""

    def display(self, head): # pylint: disable=no-self-use
        """Display all node's data."""
        current = head
        while current:
            print(current.data, end=' ')
            current = current.next

    def insert(self, head, data): # pylint: disable=no-self-use
        """Insert a node to the end of the list."""
        temp = Node(data)
        if head is None:
            head = temp
        else:
            last = head
            while last.next is not None:
                last = last.next
            last.next = temp
        return head

if __name__ == "__main__":
    MY_LIST = Solution()
    TESTS = int(input())
    HEAD = None
    for i in range(TESTS):
        DATA = int(input())
        HEAD = MY_LIST.insert(HEAD, DATA)
    MY_LIST.display(HEAD)
